### ISSUE TITLE
output on configtest

### ISSUE THREAD

**Issue by bax-** [vstakhov/rspamd#456]
When I run `rspamadm configtest`I get the following output which I don't know if it is OK:

```
2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_add_symbol: skip duplicate symbol registration for RBL_SPAMHAUS_XBL
2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; lua; ratelimit.lua:379: no servers are specified
2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; lua; dmarc.lua:291: no servers are specified for dmarc stats
2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_add_symbol: skip duplicate symbol registration for BAYES_HAM
2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_add_symbol: skip duplicate symbol registration for BAYES_SPAM
2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_validate: symbol 'DBL' has its score defined but there is no corresponding rule registered
2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_validate: symbol 'SEM_URIBL_UNKNOWN' has its score defined but there is no corresponding rule registered
2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_validate: symbol 'SEM_URIBL_FRESH15_UNKNOWN' has its score defined but there is no corresponding rule registered
syntax OK
```

**Comment by fatalbanana** [comment]
> 2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_add_symbol: skip duplicate symbol registration for RBL_SPAMHAUS_XBL

Consequence of this symbol being re-used in `composites.conf` - should be safe to ignore. May be worth doing something about.

> 2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; lua; ratelimit.lua:379: no servers are specified
> 2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; lua; dmarc.lua:291: no servers are specified for dmarc stats

Expected default behaviour. Ratelimit requires Redis to work (and in default configuration is effectively disabled) & DMARC module requires Redis only for reports (and in default configuration does not store statistics).

https://rspamd.com/doc/modules/ratelimit.html
https://rspamd.com/doc/modules/dmarc.html

> 2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_add_symbol: skip duplicate symbol registration for BAYES_HAM
> 2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_add_symbol: skip duplicate symbol registration for BAYES_SPAM

Not sure about this one but seems to be fixed in master.

> 2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_validate: symbol 'DBL' has its score defined but there is no corresponding rule registered
> 2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_validate: symbol 'SEM_URIBL_UNKNOWN' has its score defined but there is no corresponding rule registered
> 2015-12-25 18:27:23 #88600((NULL)) <qoesc1>; symcache; rspamd_symbols_cache_validate: symbol 'SEM_URIBL_FRESH15_UNKNOWN' has its score defined but there is no corresponding rule registered

Should be fixed in master as well - see #467

**Comment by bax-** [comment]
@fatalbanana Thanks a lot for your examination!
I never changed `composites.conf` so I guess this is a default config that's wrong / a duplicate.

**Comment by fatalbanana** [comment]
Yes this is indeed duplicated in the stock ruleset: apart from the annoying warning it should not be an issue.

**Comment by bax-** [comment]
Thanks!

**Comment by moisseev** [comment]
What about these messages? Are they supposed to appear on **restart**?
~~~
# service rspamd restart
Performing sanity check on rspamd configuration:
2016-01-30 11:02:58 #2813((NULL)) <z9ijff>; cfg; rspamd_rcl_symbol_handler: redefining symbol 'BAYES_SPAM' in metric 'default'
2016-01-30 11:02:58 #2813((NULL)) <z9ijff>; cfg; rspamd_rcl_symbol_handler: redefining symbol 'BAYES_HAM' in metric 'default'
~~

**Comment by fatalbanana** [comment]
That appears to be particular to your config- probably you have redefined these symbols in rspamd.conf.local- in which case I suppose this is ~fine & expected.

**Comment by moisseev** [comment]
Yes, I have them redefined in `rspamd.conf.local`. They appear on console on restart but not on start, because rc scripts don't perform configtest on start.

**Comment by rpv-tomsk** [comment]
Hi !

I have an similar warning on restart / configtest when more than one map is declared:

```
debian:/etc/rspamd# /etc/init.d/rspamd restart
[....] Restarting rapid spam filtering system: rspamd2016-08-22 14:44:51 #2953(configtest) <bmr7it>; cfg; rspamd_register_symbol_fromlua: duplicate symbol: LOCAL_SENDER_IP_WHITELIST, skip registering
. ok

debian:/etc/rspamd# rspamadm configtest
2016-08-22 14:46:57 #3438(configtest) <bmr7it>; cfg; rspamd_register_symbol_fromlua: duplicate symbol: LOCAL_SENDER_IP_WHITELIST, skip registering
syntax OK
```

The symbol declared only once, renaming it does not remove warning:

```
debian:/etc/rspamd# grep -r LOCAL_SENDER_IP_WHITELIST
local.d/multimap.conf:  LOCAL_SENDER_IP_WHITELIST {
```

The symbol itself is working correctly.

When only one map is in local.d/multimap.conf then no warning appear on restart / configtest.

What is known/fixed issue?

My rspamd version is: rspamd-1.3.3 from http://rspamd.com/apt-stable/ repository.

**Comment by fatalbanana** [comment]
@rpv-tomsk, I'm unable to replicate that. Please share `rspamadm configdump` output.

Please avoid hijacking issues. ;)

**Comment by rpv-tomsk** [comment]
Created new issue #878 for that.
